### ISSUE TITLE
Races between lkl_sys_halt and syscalls

### ISSUE THREAD

**Issue by tavip** [lkl/linux#228]
There are many possible races between lkl_sys_halt() and other threads that issue system calls:

* the syscall_thread_data structure is freed when the system call thread is terminated, but the tls key still contain a pointer to this structure
* a host thread can queue a system call to a syscall thread just before the syscall thread is be terminated leaving the host thread stuck

**Comment by liuyuan10** [comment]
I don't think this is an issue. The users should make sure all lkl syscalls are made before calling lkl_sys_halt.
